app/controllers/api/v1/jwt_api/questionnaire_controller.rb

```diff
@@ -32,13 +38,30 @@ def check_admin_authenticated
           render(status: :forbidden, json: result)
         end
 
+        def check_distributions_enabled
+          return if Rails.application.config.settings.feature_toggles.allow_distribution_export
+
+          render(status: :forbidden, json: 'Allow distribution export feature flag not enabled.')
+        end
+
         def set_questionnaire
           @questionnaire = Questionnaire.find_by(key: params[:key])
           return if @questionnaire.present?
 
           render(status: :not_found, json: 'Vragenlijst met die key niet gevonden')
         end
 
+        def set_distribution_and_check_min_responses
+          @distribution = RedisService.get("distribution_#{@questionnaire.key}") || '{}'
+
+          distribution = JSON.parse(@distribution)
+          return if distribution.present? &&
+                    distribution['total'].present? &&
+                    distribution['total'] >= Rails.application.config.settings.distribution_export_min_responses
+
+          render(status: :forbidden, json: 'Number of completed responses for this questionnaire is below the minimum')
```

Comment:
forbidden sounds like a weird response here. wouldn't 404 be better? And why not just return the `{}`?

Reply:
Good point. That would keep the front-end from breaking. If there are not enough responses, it would just show like an empty distribution instead of an error.

Reply:
Ok the vsv_api gem gets very confused if we return anything except 200, so I return an empty array with a total 0 and an error message now.